File: framework/languages/node/samples/Bingo/Server/Api/Handlers/authenticate-player-handler.js
class AuthenticatePlayerHandler {
  handle(request) {
    if (!request.accessToken?.startsWith('player-')) {
      return {
        accepted: false,
        actorId: null,
        displayName: null,
        reason: 'Access token must be a sample player id.'
      };
    }

    return {
      accepted: true,
      actorId: request.accessToken,
      displayName: request.accessToken.replace('player-', 'Player '),
      reason: null
    };
  }
}

module.exports = { AuthenticatePlayerHandler };

File: framework/languages/dotnet/samples/Bingo/Server/Api/Handlers/AuthenticatePlayerHandler.cs
using Zlink.Framework.Contracts.Handlers;
using Bingo.Shared.Contracts;

namespace Bingo.Server.Api.Handlers;

[ZLinkHandlerGroup("api")]
internal sealed class AuthenticatePlayerHandler
    : IZLinkRequestHandler<AuthenticatePlayerReq, AuthenticatePlayerRes>
{
    public ValueTask<AuthenticatePlayerRes> HandleAsync(
        AuthenticatePlayerReq request,
        ZLinkRequestContext context,
        CancellationToken cancellationToken)
    {
        cancellationToken.ThrowIfCancellationRequested();
        if (!request.AccessToken.StartsWith("player-", StringComparison.Ordinal)
            && !string.Equals(request.AccessToken, BingoSamplePlayers.Observer, StringComparison.Ordinal))
        {
            return ValueTask.FromResult(new AuthenticatePlayerRes
            {
                Accepted = false,
                Reason = "Access token must be a sample player id.",
            });
        }

        var displayName = string.Equals(request.AccessToken, BingoSamplePlayers.Observer, StringComparison.Ordinal)
            ? "Observer"
            : request.AccessToken.Replace("player-", "Player ", StringComparison.Ordinal);
        return ValueTask.FromResult(new AuthenticatePlayerRes
        {
            Accepted = true,
            ActorId = request.AccessToken,
            DisplayName = displayName,
        });
    }
}
